src/vivarium_nih_us_cvd/components/treatment.py

```diff
@@ -257,12 +263,48 @@ def on_initialize_simulants(self, pop_data: SimulantData) -> None:
         )
         pop = self.initialize_medication_coverage(pop)
 
-        # Generate outreach and polypill intervention columns
+        # Generate outreach, polypill, and lifestyle intervention columns
         # NOTE: All scenarios in this simulation start with 0%
-        # intervention exposure and so there is no need to update adherence
-        # levels at this point.
+        # intervention exposure for outreach and polypill so there
+        # is no need to update adherence levels at this point.
         pop[data_values.COLUMNS.OUTREACH] = self.outreach(pop.index)
         pop[data_values.COLUMNS.POLYPILL] = self.polypill(pop.index)
+        pop[data_values.COLUMNS.LIFESTYLE] = pd.NaT
+
+        # Generate column for last FPG test date
+        bmi = self.bmi(pop.index)
+        age = pop["age"]
+        is_eligible_for_testing = (
+            age >= data_values.FPG_TESTING.AGE_ELIGIBILITY_THRESHOLD
+        ) & (bmi >= data_values.FPG_TESTING.BMI_ELIGIBILITY_THRESHOLD)
+
+        # Determine which eligible simulants get assigned a test date
+        simulants_with_test_date = self.randomness.filter_for_probability(
+            pop[is_eligible_for_testing],
+            [data_values.FPG_TESTING.PROBABILITY_OF_TESTING_GIVEN_ELIGIBLE]
+            * sum(is_eligible_for_testing),
+        )
+
+        # sample from dates uniformly distributed from 0 to 3 years before sim start date
+        starting_event_time = self.clock()
+        fpg_test_start_time = starting_event_time
+        # Subtract step size until we reach the first day less than or equal to 3 years before our sim starting event date
+        while fpg_test_start_time >= starting_event_time - pd.DateOffset(
```

Comment:
Why are we doing this like this? Why not just draw a float from a uniform distribution, multiply that value by a `pd.TimeDelta` of 3 years, and subtract that from `self.clock()`? We don't need these times that occurred before the simulation started to occur precisely on a timestep cadence.

Reply:
I'll make this change.